main.go

```diff
@@ -167,14 +170,44 @@ func resourceManifestCreate(d *schema.ResourceData, m interface{}) error {
 	if len(data.Items) != 1 {
 		return fmt.Errorf("expected to create 1 resource, got %d", len(data.Items))
 	}
-	selflink := data.Items[0].Metadata.Selflink
-	if selflink == "" {
-		return fmt.Errorf("could not parse self-link from response %s", stdout.String())
-	}
+
+	selflink := buildSelfLink(data.Items[0].ApiVersion, data.Items[0].Metadata.Namespace, data.Items[0].Kind, data.Items[0].Metadata.Name)
```

Comment:
Though it seems to be safe, I am little not sure(or rather I am not aware how it works in real) about How it will behave when the terraform resource ID of existing resource which we generated based on selflink is now changed somwhow. It will fail to lookup the resource and recreate?

Reply:
As for existing resources, this provider reads/updates/deletes (`resourceManifestRead`, `resourceManifestUpdate`, `resourceManifestDelete`) them using existing resource ID and doesn't change ID at all as far as I read codes, so it should be ok. Of course when we recreate resources (`resourceManifestCreate`), then the new I ID we generated will be used. But I'm also little not sure how it will behave in that case and that case could happen 🤔